onnxruntime/contrib_ops/cpu/quantization/qlinear_pool.cc

```diff
@@ -247,7 +247,7 @@ struct QLinearPoolNhwc2DTask final {
     int64_t batch = begin / y_image_size;
     int64_t offset = begin % y_image_size;
 
-    for (int64_t remains = end - begin; remains > 0; offset = 0, batch++) {
+    for (int64_t remains = static_cast<int64_t>(end) - begin; remains > 0; offset = 0, batch++) {
```

Comment:
Warning 26451.  Promote it before the arithmetic to avoid overflow. However, in this case it wouldn't not overflow. So the cast is just for suppress the warning.

Reply:
For example, the following code generates the same warning:
```
int64_t foo(ptrdiff_t begin, ptrdiff_t end) {
    int64_t ret = end - begin;
    return ret;
}
```
Because ptrdiff_t could be 32 bits.